src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -1103,4 +1139,188 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
+            case BSON_OBJECT_TYPE_NAME:
+            case BSON_OBJECTID_TYPE_NAME:
+            case BSON_DBPOINTER_TYPE_NAME:
+            case BSON_MAXKEY_TYPE_NAME:
+            case BSON_MINKEY_TYPE_NAME:
+            case BSON_TIMESTAMP_TYPE_NAME:
+            case BSON_UNDEFINED_TYPE_NAME:
+            case BSON_BSON_TYPE_NAME:
+                return false;
+        }
+        throw new SQLException("unknown bson typeName: " + typeName);
+    }
+
+    protected Boolean getFixedPrecScale(String typeName) throws SQLException {
+        switch (typeName) {
+            case BSON_INT_TYPE_NAME:
+            case BSON_LONG_TYPE_NAME:
+                return true;
+            case BSON_BINDATA_TYPE_NAME:
+            case BSON_STRING_TYPE_NAME:
+            case BSON_BOOL_TYPE_NAME:
+            case BSON_DATE_TYPE_NAME:
+            case BSON_DECIMAL_TYPE_NAME:
+            case BSON_DOUBLE_TYPE_NAME:
+            case BSON_ARRAY_TYPE_NAME:
+            case BSON_OBJECT_TYPE_NAME:
+            case BSON_OBJECTID_TYPE_NAME:
+            case BSON_DBPOINTER_TYPE_NAME:
+            case BSON_JAVASCRIPT_TYPE_NAME:
+            case BSON_JAVASCRIPTWITHSCOPE_TYPE_NAME:
+            case BSON_MAXKEY_TYPE_NAME:
+            case BSON_MINKEY_TYPE_NAME:
+            case BSON_REGEX_TYPE_NAME:
+            case BSON_SYMBOL_TYPE_NAME:
+            case BSON_TIMESTAMP_TYPE_NAME:
+            case BSON_UNDEFINED_TYPE_NAME:
+            case BSON_BSON_TYPE_NAME:
+                return false;
+        }
+        throw new SQLException("unknown bson typeName: " + typeName);
+    }
+
+    protected Integer getMinScale(String typeName) throws SQLException {
+        switch (typeName) {
+            case BSON_DECIMAL_TYPE_NAME:
+                return 34;
+            case BSON_DOUBLE_TYPE_NAME:
+                return 15;
+            case BSON_BINDATA_TYPE_NAME:
+            case BSON_STRING_TYPE_NAME:
+            case BSON_INT_TYPE_NAME:
+            case BSON_BOOL_TYPE_NAME:
+            case BSON_DATE_TYPE_NAME:
+            case BSON_LONG_TYPE_NAME:
+            case BSON_ARRAY_TYPE_NAME:
+            case BSON_OBJECT_TYPE_NAME:
+            case BSON_OBJECTID_TYPE_NAME:
+            case BSON_DBPOINTER_TYPE_NAME:
+            case BSON_JAVASCRIPT_TYPE_NAME:
+            case BSON_JAVASCRIPTWITHSCOPE_TYPE_NAME:
+            case BSON_MAXKEY_TYPE_NAME:
+            case BSON_MINKEY_TYPE_NAME:
+            case BSON_REGEX_TYPE_NAME:
+            case BSON_SYMBOL_TYPE_NAME:
+            case BSON_TIMESTAMP_TYPE_NAME:
+            case BSON_UNDEFINED_TYPE_NAME:
+            case BSON_BSON_TYPE_NAME:
+                return 0;
+        }
+        throw new SQLException("unknown bson typeName: " + typeName);
+    }
+
+    protected Integer getMaxScale(String typeName) throws SQLException {
+        switch (typeName) {
+            case BSON_DECIMAL_TYPE_NAME:
+                return 34;
+            case BSON_DOUBLE_TYPE_NAME:
+                return 15;
```

Comment:
@pmeredit , @mattChiaravalloti : is this true? 
Also, I'm still not sure if we should report a max scale for a floating point decimal.

Reply:
Checked with @nbagnard and good to resolve.